R/eval.R

```diff
@@ -142,14 +139,9 @@ evaluate_call <- function(call,
                           log_warning = FALSE,
                           output_handler = new_output_handler(),
                           include_timing = FALSE) {
+  stopifnot(is.expression(exprs))
   if (debug) message(src)
 
-  if (is.null(call) && !last) {
```

Comment:
This was a performance optimisation (since we don't really need to set up all the various handlers if there's no code to execute), so removing it will make the code a little slower. However, I plan to set up all the handlers once (instead of once per top-level expression) and this makes it a little bit easier to reason about in the interim.